src/rpc_typing.c

```diff
@@ -950,7 +951,7 @@ rpct_read_event(struct rpct_file *file, struct rpct_interface *iface,
 	if (evt->result == NULL) {
 		rpc_set_last_errorf(EINVAL,
 		    "Cannot instantiate type %s of event %s",
-		    evt->result, name);
+		    (type == NULL) ? "NULL" : type, name);
```

Comment:
if `type` is null then `rpct_read_event()` should bail out earlier.

Reply:
Yeah, I was wondering why the previous line checks for type before calling rpct_instantiate_type - it looked like there was an intent to capture both errors at the same same. Do you want an error message for a null type, or for it to be silently ignored?

Reply:
`type == NULL` means that the user didn't specify a type in the .yaml file and it should be treated like syntax error.

Reply:
Done